app/components/filters/text-filter.ts

```diff
@@ -1,14 +1,29 @@
 import FilterComponent from './filter';
 import { action } from '@ember/object';
+import { localCopy } from 'tracked-toolbox';
+
+export interface Signature {
+  Args: {
+    id: string;
+    queryParam: string;
+    value: string;
+  };
+}
+
+export default class TextFilterComponent extends FilterComponent<Signature> {
+  @localCopy<string>('args.value', '') declare value: string;
```

Comment:
This allows us to use a local 'value' property that still updates when the argument changes. We don't want to update the argument directly as that would update the QP and trigger the model hook.

Alternatively we could still update the QP value directly but then we would have to remove the `refreshModel` value and trigger the update with `this.router.refresh('route-name')`. The downside of that is that if a user types in the field  and doesn't submit, and refreshes the page, the value would still be there which seemed weird since they didn't press search yet. So I opted for the localCopy version.